### ISSUE TITLE
Container view don't update when using focus mode

### ISSUE THREAD

**Issue by ThomasDucroquetzDax** [deneb-viz/deneb#481]
Hi,
I recently converted a vegaLite chart to vega in order to create a more dynamic chart.
The chart is a simple bar chart with some customizations.
I noticed a potential issue in **Power BI Service** when I use **focus mode**. 
When I use Power BI Desktop, no problem, my chart is responsive and use the full width and height.

![image](https://github.com/user-attachments/assets/f23fe281-931c-4e96-b74e-9d804e658709)

But when I am in Power BI Service and only in view mode (not dev mode), my chart just doesn't adjust and keep the same width and height as the initial chart which leaves a huge blank.

![image](https://github.com/user-attachments/assets/b56c76e9-1fdb-4149-86e4-883e1026fcd5)

To debug that issue, I inserted a text mark on the top of my chart with : pbiContainerWidth | width | pbiContainer.width. As you can see those variables does update in dev mode, but not in view mode.

It seems that this behavior is common on all my vega charts but not vega-lite. I was wondering if this behavior concerns the new 1.7 version or this has to do with my json spec.

I've joined my spec  (there are a bunch of debug marks and signal)
[VegaSpec.json](https://github.com/user-attachments/files/16430909/VegaSpec.json)

Here is the config part :
```
{
  "autosize": {
    "type": "fit-x",
    "contains": "padding",
    "resize": true
  },
  "padding": {
    "left": 0,
    "top": 0,
    "right": 0,
    "bottom": 0
  },
  "view": {"stroke": "black"},
  "font": "Segoe UI,Segoe UI_MSFontService,sans-serif"
}
``` 

Thank you very much for your support

**Comment by ThomasDucroquetzDax** [comment]
I did some additional tests with minimal configuration and it seems that this behavior is specific to vega.

I've created a simple report with two bar charts : 

- one in Vega
- on in Vega-lite

The Vega and Vega-Lite charts will scale in focus mode from Power BI Desktop and Power BI Service dev mode.
Only the Vega-Lite Chart scales in focus mode from Power BI Service view mode.

I've also tested with Deneb 1.6.2 (standalone), I have the same behavior.

Thank you very much for your support and for this amazing visual.

Here is the vega spec : 
```
{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "description": "A basic bar chart example, with value labels shown upon pointer hover.",
  "padding": 5,
  "autosize": {
    "type": "fit",
    "contains": "padding"
  },
  "data": [
    {
      "name": "table",
      "values": [
        {"category": "A", "amount": 28},
        {"category": "B", "amount": 55},
        {"category": "C", "amount": 43},
        {"category": "D", "amount": 91},
        {"category": "E", "amount": 81},
        {"category": "F", "amount": 53},
        {"category": "G", "amount": 19},
        {"category": "H", "amount": 87}
      ]
    }
  ],

  "signals": [
    {
      "name": "tooltip",
      "value": {},
      "on": [
        {"events": "rect:pointerover", "update": "datum"},
        {"events": "rect:pointerout",  "update": "{}"}
      ]
    }
  ],

  "scales": [
    {
      "name": "xscale",
      "type": "band",
      "domain": {"data": "table", "field": "category"},
      "range": "width",
      "padding": 0.05,
      "round": true
    },
    {
      "name": "yscale",
      "domain": {"data": "table", "field": "amount"},
      "nice": true,
      "range": "height"
    }
  ],

  "axes": [
    { "orient": "bottom", "scale": "xscale" },
    { "orient": "left", "scale": "yscale" }
  ],

  "marks": [
    {
      "type": "rect",
      "from": {"data":"table"},
      "encode": {
        "enter": {
          "x": {"scale": "xscale", "field": "category"},
          "width": {"scale": "xscale", "band": 1},
          "y": {"scale": "yscale", "field": "amount"},
          "y2": {"scale": "yscale", "value": 0}
        },
        "update": {
          "fill": {"value": "steelblue"}
        },
        "hover": {
          "fill": {"value": "red"}
        }
      }
    },
    {
      "type": "text",
      "encode": {
        "enter": {
          "align": {"value": "center"},
          "baseline": {"value": "bottom"},
          "fill": {"value": "#333"}
        },
        "update": {
          "x": {"scale": "xscale", "signal": "tooltip.category", "band": 0.5},
          "y": {"scale": "yscale", "signal": "tooltip.amount", "offset": -2},
          "text": {"signal": "tooltip.amount"},
          "fillOpacity": [
            {"test": "datum === tooltip", "value": 0},
            {"value": 1}
          ]
        }
      }
    }
  ]
}
``` 

Here is the Vega-Lite spec : 
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A simple bar chart with embedded data.",
  "padding": 5,
  "autosize": {
    "type": "fit",
    "contains": "padding"
  },
  "data": {
    "values": [
      {"a": "A", "b": 28}, {"a": "B", "b": 55}, {"a": "C", "b": 43},
      {"a": "D", "b": 91}, {"a": "E", "b": 81}, {"a": "F", "b": 53},
      {"a": "G", "b": 19}, {"a": "H", "b": 87}, {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "nominal", "axis": {"labelAngle": 0}},
    "y": {"field": "b", "type": "quantitative"}
  }
}
```

**Comment by dm-p** [comment]
Hi @ThomasDucroquetzDax, and thanks for raising. I appreciate the detailed steps to help me reproduce also 👍

I'm a bit short on available time for Deneb support this week, but I have had a very quick look at the behavior. This is partially to confirm your acknowledgment of your issue and to provide me with some notes for when I can pick this up.

The Vega issue is indeed not present in Desktop, but it is not present in the Service when the report is in **Edit** mode. It works as expected here.

The issue is present for reading mode (which, let's face it, will be how most people consume the visual).

I suspect that because the Desktop is perpetually in edit mode, and we have to have a report open in edit mode in the Service to develop and debug, there is a crucial difference in the updates that happen when a visual is set into focus mode.

For background, when Power BI transitions the visual to and from focus mode, it makes several updates to the container dimensions during this process, and we try to catch the first and last step of this to ensure that we don't keep recalculating the visual during this time. We also use this to ensure that we keep the visual the "view mode" size of the visual when editing, so that it represents how it should look on the canvas.

This pattern might be different in reading mode in the Service, so we will need to analyze it and implement a suitable check. I'm planning to make this much better in 2.0, but as I don't have a solid ETA, I think we should look at doing a remediation earlier than this (probably 1.8).

**Comment by ThomasDucroquetzDax** [comment]
Hi @dm-p, thank you very much for your answer and I'm glad you will check and try to fix it for 1.8.
Your explanation makes sense and if you need anything to help you debug the issue, I will be glad to help you.
Thanks again for your devotion and for your amazing work.

**Comment by dm-p** [comment]
Hi @ThomasDucroquetzDax - I need to do a critical fix for tooltips, so am assembling a v1.7.1. Your issue should be a quick win, and I can fix it as part of this update if I have understood the root cause correctly. If you can, please test this build for the use cases you've identified and confirm that the behavior is as expected. If so, I can include this in the AppSource submission.

* Download the file
* Remove the .zip extension (so it is now a valid .pbiviz)
* Load into your workbook and convert your affected visual over to this alpha version
* Confirm the viewport changes as expected for Vega and Vega-Lite visuals, i.e.:
    * Editor opens with visual in correct size/aspect ratio in the preview area
    * Setting the visual into focus mode uses the full viewport
    * This also behaves as expected in the Power BI Service

This alpha also contains all fixes [currently scheduled for 1.7.1](https://github.com/deneb-viz/deneb/milestone/24), so if you have any further thoughts on those, you are welcome to provide feedback on those issues accordingly also :)

If you encounter any further issues, please provide reproducible examples, as you have been doing. Your efforts in this regard are greatly appreciated and significantly speed up the isolation and fixing process. Thank you for your continued support 🙏 

File: [deneb-alpha.1.7.1.20240801#ef7c939.pbiviz.zip](https://github.com/user-attachments/files/16450379/deneb-alpha.1.7.1.20240801.ef7c939.pbiviz.zip)

> The usual disclaimers apply about downloading visuals from unsolicited sources. I attest that while this alpha build uses a different GUID and is therefore regarded as uncertified by Power BI, it is built with certified visual permissions (no external comms). It conforms to [Deneb's privacy policy](https://deneb-viz.github.io/privacy-policy) **If downloaded directly from this above link**.

**Comment by ThomasDucroquetzDax** [comment]
Hi @dm-p,
Thanks for following up.

I've tested all use cases I mentioned above and all works as expected, well done 👍

While testing, I found another potential issue when scrolling and interacting with the visual, not sure if this is related to issue #475. I found out that when the visual is scrolled and that I click on my visual to cross filter other visuals, the visual is resetted to original position (top in my case)

Do you want me to raise another issue to share more or are you aware of this behavior ? This seems to be the case on 1.7 and 1.7.1 alpha, didn't tested on 1.6.2 or other builds.

Again, thanks a lot for what you are doing

**Comment by dm-p** [comment]
Awesome to hear it's working for you! I'll ensure this is included in the 1.7.1 submission when we've closed off the other issues. The standalone build will be available ahead of this, and I'll confirm here when it's ready.

The other issue you've identified is already being tracked under #374. This will require a much more significant change (which I alluded to above), and we'll address it in the work for 2.0.
